### ISSUE TITLE
`newline-after-import` deosn't take care of comments

### ISSUE THREAD

**Issue by manuth** [import-js/eslint-plugin-import#2151]
The `newline-after-import` rule doesn't hit if the import-statements are immediately followed by comments:

This piece of code is expected to produce an error, but it doesn't:
```ts
import { join } from "path";
/**
 *
 */
console.log();
```

**Comment by ljharb** [comment]
From where do you form this expectation? The docs don't have any examples with comments, nor do [the tests](https://github.com/benmosher/eslint-plugin-import/blob/master/tests/src/rules/newline-after-import.js).

The goal of the rule is to ensure that there's some spacing between imports and non-imports - arguably a comment suffices.

**Comment by alvis** [comment]
I come to this issue for the exact same problem. In the OP's case, I'd expect
```ts
import { join } from "path";

/**
 *
 */
console.log();
```

The examples and tests may not include comment into consideration but I find it more intuitive and more clear if there's an empty line after all the import statements. After all, I believe it's the intention of the rule. For example, the following annoying example doesn't make sense to me but it's a pass today:
```ts
import something from 'somewhere';
// it's a very looooooooooooooooooooooooooooooooooooog comment
console.log('is it a good quality code?')
```

**Comment by ljharb** [comment]
A PR to add an option to make comments not be counted as an empty line would be appreciated.

**Comment by SrBrahma** [comment]
Coming for the same reason. Don't have knowledge with eslint plugins nor the time to get into it right now.

**Comment by pri1311** [comment]
Hey! I would like to work on this.

**Comment by ljharb** [comment]
Go for it!
